sparse/rest/handlers.go

```diff
@@ -0,0 +1,166 @@
+package rest
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancher/sparse-tools/sparse"
+)
+
+func (server *SyncServer) sendErrorResponse(errorMessage string, writer http.ResponseWriter) {
+	log.Error(errorMessage)
+	http.Error(writer, errorMessage, http.StatusInternalServerError)
+}
+
+func (server *SyncServer) getQueryInterval(request *http.Request) (sparse.Interval, error) {
+	var interval sparse.Interval
+	var err error
+
+	queryParams := request.URL.Query()
+	if len(queryParams["begin"]) == 0 || len(queryParams["end"]) == 0 {
+		err = fmt.Errorf("queryParams begin or end not exist")
+		return interval, err
+	}
+
+	begin, err := strconv.ParseInt(queryParams["begin"][0], 10, 64)
+	if err != nil {
+		err = fmt.Errorf("strconv.ParseInt(begin) error: %s", err)
+		return interval, err
+	}
+	end, err := strconv.ParseInt(queryParams["end"][0], 10, 64)
+	if err != nil {
+		err = fmt.Errorf("strconv.ParseInt(end) error: %s", err)
+		return interval, err
+	}
+
+	return sparse.Interval{begin, end}, err
+}
+
+func (server *SyncServer) open(writer http.ResponseWriter, request *http.Request) {
+	// get file size
+	interval, err := server.getQueryInterval(request)
+	if err != nil {
+		errorMessage := fmt.Sprintf("server.getQueryInterval failed, err: %s", err)
```

Comment:
It's not about one line shorter or not.

It's normal for function to return err rather than calling `server.sendErrorResponse` every time when you caught a error.

But since the http handler has no error return, so it make sense for it to call a function with error return and deal with it in one place, rather than calling `server.sendErrorResponse` every time it got a error.

The callings to `server.sendErrorResponse` should be in one place, rather than scattering in the every step of handling.